config/config/src/main/java/io/helidon/config/internal/UrlConfigSource.java

```diff
@@ -155,9 +158,25 @@ private String mediaType(String responseMediaType) {
         return mediaType;
     }
 
-    private String probeContentType() {
-        String path = url.getPath();
-        return ConfigHelper.detectContentType(Paths.get(path));
+    private String probeContentType() throws URISyntaxException {
+        URI uri = url.toURI();
+        Path path;
+        switch (uri.getScheme()) {
+            case "jar":
+                String relativePath = uri.getSchemeSpecificPart();
+                int idx = relativePath.indexOf("!");
+                if (idx > 0) {
+                    relativePath = relativePath.substring(idx + 1);
```

Comment:
Maybe make sure that `relativePath`'s length will actually support this operation.  So perhaps `if (idx > 0 && relativePath.length() >= idx + 1)` or something like that